client.go

```diff
@@ -362,213 +108,152 @@ func (client *Client) AddWithRetention(key string, timestamp int64, value float6
-	values, err := redis.Values(info, err)
-	if err != nil {
-		return nil, err
-	}
-	if len(values) == 0 {
-		return []DataPoint{}, nil
-	}
-	for _, i := range values {
-		iValues, err := redis.Values(i, err)
-		if err != nil {
-			return nil, err
-		}
-		rawTimestamp := iValues[0]
-		strValue, err := redis.String(iValues[1], nil)
-		if err != nil {
-			return nil, err
-		}
-		timestamp, err := redis.Int64(rawTimestamp, nil)
-		if err != nil {
-			return nil, err
-		}
-		value, err := strToFloat(strValue)
-		if err != nil {
-			return nil, err
-		}
-		dataPoint := DataPoint{timestamp, value}
-		dataPoints = append(dataPoints, dataPoint)
-	}
-	return dataPoints, nil
-}
-
-func ParseLabels(res interface{}) (labels map[string]string, err error) {
-	values, err := redis.Values(res, err)
-	if err != nil {
-		return
-	}
-	labels = make(map[string]string, len(values))
-	for i := 0; i < len(values); i++ {
-		iValues, err := redis.Values(values[i], err)
-		if err != nil {
-			return nil, err
-		}
-		if len(iValues) != 2 {
-			err = errors.New("ParseLabels: expects 2 elements per inner-array")
-			return nil, err
-		}
-		key, okKey := iValues[0].([]byte)
-		value, okValue := iValues[1].([]byte)
-		if !okKey || !okValue {
-			err = errors.New("ParseLabels: StringMap key not a bulk string value")
-			return nil, err
-		}
-		labels[string(key)] = string(value)
-	}
-	return
-}
-
-type Range struct {
-	Name       string
-	Labels     map[string]string
-	DataPoints []DataPoint
+// TS.CREATERULE create a compaction rule
+// SOURCE_KEY - key name for source time series
+// AGG_TYPE - AggregationType
+// BUCKET_SIZE_SEC - time bucket for aggregated compaction,
+// DEST_KEY - key name for destination time series
+func (client *Client) CreateRule(sourceKey string, aggType AggregationType, bucketSizeSec uint, destinationKey string) (err error) {
+	conn := client.Pool.Get()
+	defer conn.Close()
+	_, err = conn.Do("TS.CREATERULE", sourceKey, destinationKey, "AGGREGATION", aggType, bucketSizeSec)
+	return err
 }
 
-func ParseRanges(info interface{}) (ranges []Range, err error) {
-	values, err := redis.Values(info, err)
-	if err != nil {
-		return nil, err
-	}
-	if len(values) == 0 {
-		return []Range{}, nil
-	}
-
-	for _, i := range values {
-		iValues, err := redis.Values(i, err)
-		if err != nil {
-			return nil, err
-		}
-		if len(iValues) != 3 {
-			err = errors.New("ParseRanges: expects 3 elements per inner-array")
-			return nil, err
-		}
-
-		name, err := redis.String(iValues[0], nil)
-		if err != nil {
-			return nil, err
-		}
-
-		labels, err := ParseLabels(iValues[1])
-		if err != nil {
-			return nil, err
-		}
-
-		dataPoints, err := ParseDataPoints(iValues[2])
-		if err != nil {
-			return nil, err
-		}
-		r := Range{name, labels, dataPoints}
-		ranges = append(ranges, r)
-	}
-	return ranges, nil
+// DeleteRule - delete a compaction rule
+// args:
+// SOURCE_KEY - key name for source time series
+// DEST_KEY - key name for destination time series
```

Comment:
Args style

Reply:
Fixing on new commit and revising all methods description